contrib/jax_ws/src/python/pants/contrib/jax_ws/tasks/jax_ws_gen.py

```diff
@@ -0,0 +1,91 @@
+# coding=utf-8
+# Copyright 2017 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import (absolute_import, division, generators, nested_scopes, print_function,
+                        unicode_literals, with_statement)
+
+import logging
+import os
+
+from pants.backend.jvm.targets.java_library import JavaLibrary
+from pants.backend.jvm.tasks.nailgun_task import NailgunTask
+from pants.base.exceptions import TaskError
+from pants.java.distribution.distribution import DistributionLocator
+from pants.task.simple_codegen_task import SimpleCodegenTask
+
+from pants.contrib.jax_ws.targets.jax_ws_library import JaxWsLibrary
+
+logger = logging.getLogger(__name__)
+
+
+class JaxWsGen(SimpleCodegenTask, NailgunTask):
+  """Generates Java files from wsdl files using the JAX-WS compiler."""
+
+  @classmethod
+  def register_options(cls, register):
+    super(JaxWsGen, cls).register_options(register)
+    register('--ws-quiet', type=bool, help='Suppress WsImport output')
+    register('--ws-verbose', type=bool, help='Make WsImport output verbose')
+
+  @classmethod
+  def subsystem_dependencies(cls):
+    return super(JaxWsGen, cls).subsystem_dependencies() + (DistributionLocator,)
+
+  def __init__(self, *args, **kwargs):
+    super(JaxWsGen, self).__init__(*args, **kwargs)
+
+  def synthetic_target_type(self, target):
+    return JavaLibrary
+
+  def is_gentarget(self, target):
+    return isinstance(target, JaxWsLibrary)
+
+  @classmethod
+  def supported_strategy_types(cls):
+    return [cls.IsolatedCodegenStrategy]
+
+  def execute_codegen(self, target, target_workdir):
+    distribution = DistributionLocator.cached(jdk=True)
+    classpath = distribution.find_libs(['tools.jar'])
+
+    wsdl_directory = target.payload.sources.rel_path
+    for source in target.payload.sources.source_paths:
+      url = os.path.join(wsdl_directory, source)
+
+      args = self._format_args_for_relative_path(target, target_workdir, url)
+
+      result = self.runjava(
+        classpath=classpath,
+        main='com.sun.tools.internal.ws.WsImport',
+        jvm_options=target.payload.vm_args,
```

Comment:
If targets have different vm_args, this will result in the nailgun'd JVM being torn down on a target by target basis... is it expected that all targets will set a particular set of options? If so, it might be better to just have people configure them for their repo using `[jvm.gen.jax-ws]` in `pants.ini`.